src/TCommandLine.cpp

```diff
@@ -279,192 +212,227 @@ bool TCommandLine::event( QEvent * event )
-                }
-                else
-                    mHistoryBuffer = 0;
-                adjustHeight();
+            }
+            QPlainTextEdit::event(event);
+            adjustHeight();
+            return true;
+            break;
+
+        case Qt::Key_Return:
+            if (ke->modifiers() & Qt::ControlModifier) {
+                mpConsole->console2->mCursorY = mpConsole->buffer.size();
+                mpConsole->console2->hide();
+                mpConsole->buffer.mCursorY = mpConsole->buffer.size();
+                mpConsole->console->mCursorY = mpConsole->buffer.size();
+                mpConsole->console->mIsTailMode = true;
+                mpConsole->console->updateScreenView();
+                mpConsole->console->forceUpdate();
                 ke->accept();
                 return true;
-                break;
-
-            case Qt::Key_Down:
-                if( ke->modifiers() & Qt::ControlModifier )
-                {
-                    moveCursor(QTextCursor::Down, QTextCursor::MoveAnchor);
-                    ke->accept();
-                    return true;
-                }
-                else
-                {
-                    historyDown(ke);
-                    ke->accept();
-                    return true;
-                }
-                break;
-
-            case Qt::Key_Up:
-                if( ke->modifiers() & Qt::ControlModifier )
-                {
-                    moveCursor(QTextCursor::Up, QTextCursor::MoveAnchor );
-                    ke->accept();
-                    return true;
-                }
-                else
-                {
-                    historyUp(ke);
-                    ke->accept();
-                    return true;
-                }
-                break;
-
-            case Qt::Key_Escape:
-
-                selectAll();
-                mAutoCompletion = false;
-                mTabCompletion = false;
+            } else if (ke->modifiers() & Qt::ShiftModifier) {
+                textCursor().insertBlock();
+                ke->accept();
+                return true;
+            } else {
+                enterCommand(ke);
+                mTabCompletionCount = -1;
+                mAutoCompletionCount = -1;
+                mLastCompletion = "";
                 mTabCompletionTyped = "";
                 mAutoCompletionTyped = "";
                 mUserKeptOnTyping = false;
                 mTabCompletionCount = -1;
                 mAutoCompletionCount = -1;
-                setPalette( mRegularPalette );
-                if( mpHost->mAutoClearCommandLineAfterSend )
+                if (mpHost->mAutoClearCommandLineAfterSend) {
+                    clear();
                     mHistoryBuffer = -1;
-                else
+                } else
                     mHistoryBuffer = 0;
+                adjustHeight();
                 ke->accept();
                 return true;
-                break;
+            }
+            break;
```

Comment:
All execution routes do `ke->accept();  return true;` so could be outside conditional branches - and the `break;` remains unreachable...!